Examples/Python/tests/test_examples.py

```diff
@@ -542,3 +585,123 @@ def test_ckf_tracks_example_truth_smeared(tmp_path):
     assert len([f for f in csv.iterdir() if f.name.endswith("CKFtracks.csv")]) == events
     assert all([f.stat().st_size > 300 for f in csv.iterdir()])
+
+
+@pytest.mark.skipif(not dd4hepEnabled, reason="DD4hep not set up")
+@pytest.mark.slow
+# @pytest.mark.filterwarnings("ignore::UserWarning")
+def test_vertex_fitting(tmp_path):
+    detector, trackingGeometry, decorators = getOpenDataDetector()
+
+    field = acts.ConstantBField(acts.Vector3(0, 0, 2 * u.T))
+
+    from vertex_fitting import runVertexFitting, VertexFinder
+
+    s = Sequencer(events=10)
+
+    runVertexFitting(
+        field,
+        vertexFinder=VertexFinder.Truth,
+        outputDir=Path.cwd(),
+        s=s,
+    )
+
+    alg = AssertCollectionExistsAlg(["fittedVertices"], name="check_alg")
+    s.addAlgorithm(alg)
+
+    if isCI:
+        with pytest.raises(RuntimeError, match=".*ACTS_LOG_FAILURE_THRESHOLD.*"):
+            s.run()
+    else:
+        s.run()
+        assert alg.events_seen == s.config.events
+
+
+import itertools
+
+
+@pytest.mark.parametrize(
+    "finder,inputTracks,entries",
+    [
+        ("Truth", False, 1),
+        # ("Truth", True, 0), # this combination seems to be not working
+        ("Iterative", False, 1),
+        ("Iterative", True, 10),
+        ("AMVF", False, 1),
+        ("AMVF", True, 10),
+    ],
+)
+@pytest.mark.filterwarnings("ignore::UserWarning")
+def test_vertex_fitting_reading(tmp_path, ptcl_gun, rng, finder, inputTracks, entries):
+
+    ptcl_file = tmp_path / "particles.root"
+
+    detector, trackingGeometry, decorators = GenericDetector.create()
+    field = acts.ConstantBField(acts.Vector3(0, 0, 2 * u.T))
+
+    from vertex_fitting import runVertexFitting, VertexFinder
+
+    inputTrackSummary = None
+    if inputTracks:
+        from truth_tracking import runTruthTracking
+
+        s2 = Sequencer(numThreads=1, events=10)
+        runTruthTracking(
+            trackingGeometry,
+            field,
+            digiConfigFile=Path(
+                "Examples/Algorithms/Digitization/share/default-smearing-config-generic.json"
+            ),
+            outputDir=tmp_path,
+            s=s2,
+        )
+        s2.run()
+        del s2
+        inputTrackSummary = tmp_path / "tracksummary_fitter.root"
+        assert inputTrackSummary.exists()
+        assert ptcl_file.exists()
+    else:
+        s0 = Sequencer(events=1)
+        evGen = ptcl_gun(s0)
+        s0.addWriter(
+            RootParticleWriter(
+                level=acts.logging.INFO,
+                inputParticles=evGen.config.outputParticles,
+                filePath=str(ptcl_file),
+            )
+        )
+        s0.run()
+        del s0
+
+        assert ptcl_file.exists()
+
+    finder = VertexFinder[finder]
+
+    s3 = Sequencer(numThreads=1)
+
+    runVertexFitting(
+        field,
+        inputParticlePath=ptcl_file,
+        inputTrackSummary=inputTrackSummary,
+        outputDir=tmp_path,
+        vertexFinder=finder,
+        s=s3,
+    )
```

Comment:
Do we want to had a test with ptcl_file = NONE ? Or do you thig this is not needed ?

Reply:
I think we should actually have this one.